### PR TITLE
BEM-2551: bem-history: Не переходит назад в Crome в iOS

### PR DESCRIPTION
**Pull request by baymer**

### CODE REVIEW

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 1]
```diff
@@ -55,8 +55,13 @@ provide(inherit(Base, /** @lends history.prototype */{
 
     _syncState : function() {
         // Replace null with undefined to catch initial popstate
-        if(window.history.state === null) {
-            window.history.replaceState(undefined, document.title, window.location.href);
+        // По спецификации первоначальное значение state должно равняться null.
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: English, please

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 2]
```diff
@@ -55,8 +55,13 @@ provide(inherit(Base, /** @lends history.prototype */{
...
+        // Но в браузере UC для Android history.state всегда undefined
+        if (state === null || typeof state === 'undefined') {
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: shouldn't it be `window.history.state` instead?

Comment (baymer) [inline common.blocks/history/_provider/history_provider_history-api.js]: @tadatuta, fixed

File: common.blocks/history/_provider/history_provider_history-api.js (hunk 1 above)
Comment (baymer) [inline common.blocks/history/_provider/history_provider_history-api.js]: Done

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 3]
```diff
@@ -54,10 +54,18 @@ provide(inherit(Base, /** @lends history.prototype */{
...
 
     _syncState : function() {
-        // Replace null with undefined to catch initial popstate
-        if(window.history.state === null) {
-            window.history.replaceState(undefined, document.title, window.location.href);
+        var state = window.history.state;
+
+        // Replace null with undefined to catch initial popstate.
+        // Initial val must be null (in specification).
+        // But in the UC browser for Android history.state equals null.
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: looks like there's an error in comments (null in spec and in UC now)

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 4]
```diff
@@ -54,10 +54,18 @@ provide(inherit(Base, /** @lends history.prototype */{
...
+        if (state === null || typeof state === 'undefined') {
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: please run linter before commit (see https://travis-ci.org/bem/bem-history/jobs/117659145 for details).

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 5]
```diff
@@ -54,10 +54,18 @@ provide(inherit(Base, /** @lends history.prototype */{
...
+            // In the Chrome browser for iOS when history.state
+            // replaced to undefined the popstate event will not be triggered.
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: replaced to -> replaced with

File: common.blocks/history/_provider/history_provider_history-api.js (hunk 3 above)
Comment (baymer) [inline common.blocks/history/_provider/history_provider_history-api.js]: Yep, I will fix

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 6]
```diff
@@ -54,10 +54,18 @@ provide(inherit(Base, /** @lends history.prototype */{
...
 
     _syncState : function() {
-        // Replace null with undefined to catch initial popstate
-        if(window.history.state === null) {
-            window.history.replaceState(undefined, document.title, window.location.href);
+        var state = window.history.state;
+
+        // Replace null with undefined to catch initial popstate.
+        // Initial val must be null (in specification).
+        // But in the UC browser for Android history.state always is undefined.
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: always is  -> is always

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 7]
```diff
@@ -54,10 +54,18 @@ provide(inherit(Base, /** @lends history.prototype */{
...
+        if(state === null || typeof state === 'undefined') {
+            // In the Chrome browser for iOS when history.state
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js:63]: when history.state **is** replaced

File: common.blocks/history/_provider/history_provider_history-api.js [hunk 8]
```diff
@@ -54,10 +54,18 @@ provide(inherit(Base, /** @lends history.prototype */{
     },
 
     _syncState : function() {
-        // Replace null with undefined to catch initial popstate
-        if(window.history.state === null) {
-            window.history.replaceState(undefined, document.title, window.location.href);
+        var state = window.history.state;
+
+        // Replace null with undefined to catch initial popstate.
```
Comment (tadatuta) [inline common.blocks/history/_provider/history_provider_history-api.js]: let's rewrite it as `Replace null with empty object to catch initial popstate`

**Comment by tadatuta** [comment]
lgtm, will you, please, squash commits?

**Comment by tadatuta** [comment]
@baymer thank you!

**Pull request merged**